### PR TITLE
feat: Add static error page

### LINKED ISSUE #4040: Custom proxy error page
When proxying, instead of rendering the dashboard with an error page we should render a small non-SPA HTML file that contains the error message and a button to refresh, a button to go to dashboard (controlled by Go templates).

We have this in v1 for when you get proxy errors. It can be reproduced by creating a dev url and trying to connect to it without starting an app on that port first:
image

## Requirements
1. Completely self-contained page, with no `<link rel="stylesheet" ...>`s or `<script src="..."/>`s
2. Go template variables for:
    - status code `Error.Status`
    - error title `Error.Title`
    - error description `Error.Description`
    - optional button to reload based on `Error.Reloadable`
    - button to redirect to dashboard `Error.DashboardURL`
3. Stored at `site/static/simple-error.html` or something like that

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Screen Shot 2022-09-30 at 13 41 54

### CODE REVIEW

**Review by jsjoeio — APPROVED** [review]
Nice work!

File: site/static/error.html [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            />
+          </clipPath>
+        </defs>
+      </svg>
+
+      <h1>502 - Bad Gateway</h1>
+      <p>No service is running on port 3000. Make sure your process is running and try again.</p>
+      <div class="button-group">
+        <button id="retry">Retry</button>
+        <a href="https://dev.coder.com"> Back to site </a>
```
Comment (kylecarbs) [inline site/static/error.html:126]: This can't be static ;p

File: site/static/error.html [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+              transform="translate(0.902344 5.71326)"
+            />
+          </clipPath>
+        </defs>
+      </svg>
+
+      <h1>502 - Bad Gateway</h1>
+      <p>No service is running on port 3000. Make sure your process is running and try again.</p>
+      <div class="button-group">
+        <button id="retry">Retry</button>
```
Comment (kylecarbs) [inline site/static/error.html:125]: Will this button be dynamic?

File: site/static/error.html (hunk 1 above)
Comment (BrunoQuaresma) [reply site/static/error.html:126]: Yes, @deansheather is going to replace those values with Go templates. I'm confirming with him.

File: site/static/error.html (hunk 2 above)
Comment (BrunoQuaresma) [reply site/static/error.html:125]: Probably. I think this is something that Go template can handle right?

**Comment by deansheather** [comment]
Added the go template variables and made a few changes:
- Added `justify-content: center` to `.button-group` so the return to site button would be centered when the retry button was not present
- Inlined the `window.location.reload()` to avoid making the JS conditional as well

File: site/static/error.html (hunk 2 above)
Comment (deansheather) [reply site/static/error.html]: Yes, controlled by `.Error.RetryEnabled`

**Review by deansheather — APPROVED** [review]

**Pull request merged**